holmes/plugins/toolsets/redis.py

```
@@ -0,0 +1,484 @@
+            if "redis://" in redis_url or "rediss://" in redis_url:
+                client = redis.from_url(redis_url, db=selected_db, password=password)  # Use selected_db
+            else:
+                host, port_str = redis_url.split(":")
+                port = int(port_str)
+                client = redis.Redis(host=host, port=port, db=selected_db, password=password) # Use selected_db
+
+            client.ping()
+            return client
+        except redis.exceptions.ConnectionError as e:
+            logging.exception("Failed to connect to Redis")
+            raise ConnectionError(f"Could not connect to Redis: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)